python/pyspark/streaming/dstream.py

```diff
@@ -579,9 +580,9 @@ def reduceFunc(t, a, b):
                 g = b.groupByKey(numPartitions).mapValues(lambda vs: (list(vs), None))
             else:
                 g = a.cogroup(b.partitionBy(numPartitions), numPartitions)
-                g = g.mapValues(lambda (va, vb): (list(vb), list(va)[0] if len(va) else None))
```

Comment:
This [Python 3 change](https://www.python.org/dev/peps/pep-3113/) is annoying, but I understand the rationale, especially after witnessing users getting `lambda x, y` and `lambda (x, y)` confused.  If we use the Python 2.x tuple unpacking syntax in our examples, then I guess we should be prepared for some users to get confused when our Python 2.x examples in the programming guide lead to syntax errors in Python 3.  We can deal with these issues as they arise, though.